pkg/handler/unrouted_handler.go

```diff
@@ -338,7 +340,8 @@ func (handler *UnroutedHandler) PostFile(w http.ResponseWriter, r *http.Request)
 	}
 
 	if handler.config.PreUploadCreateCallback != nil {
-		resp2, changes, err := handler.config.PreUploadCreateCallback(newHookEvent(c, info))
+		access := newAccessInfo(AccessModeRead, partialFileInfos)
+		resp2, changes, err := handler.config.PreUploadCreateCallback(newHookEvent(c, &info, &access))
```

Comment:
For concatenation, I think we should first emit a pre-access hook to check the permissions for read access on the partial uploads. If that succeeds we can continue with the upload creation and the corresponding pre-create hook. Then users don't have to duplicate access-related logic in pre-access and pre-create.

A good location for that hook is likely a few lines above after the call to `handler.sizeOfUploads`.